src/repr/row.rs

```diff
@@ -295,19 +295,23 @@ unsafe fn read_datum<'a>(data: &'a [u8], offset: &mut usize) -> Datum<'a> {
 // --------------------------------------------------------------------------------
 // writing data
 
-fn assert_is_copy<T: Copy>(_t: T) {}
+fn assert_is_plain_data<T: Copy + 'static>(_t: T) {}
 
 // See https://github.com/rust-lang/rust/issues/43408 for why this can't be a function
-macro_rules! push_copy {
+//
+// Asserts that a value is "plain data" and then pushes its bytes onto a buffer.
+// "Plain data" means Copy (i.e., can validly be transmuted to and from [u8; _])
```

Comment:
`Copy` is necessary but not sufficient eg if `T` contains padding bytes then it's UB to convert to `[u8;_]`. Unfortunately there is no way to check that in the type system so we rely on the miri tests at the end of the file.